### DISCUSSION TITLE
use shiki decorations in nuxt content

### DISCUSSION THREAD

**Discussion by dbzx10299** [nuxt/content#2554] [Q&A]
In the shiki docs it gives an example of how to use decorations
```js
const html = await codeToHtml(code, {
  theme: 'vitesse-light',
  lang: 'ts',
  decorations: [ 
    {
      // line and character are 0-indexed
      start: { line: 1, character: 0 },
      end: { line: 1, character: 11 },
      properties: { class: 'highlighted-word' }
    }
  ]
})
```

If I have a custom prose pre component i need to be able to generate a decoration object and push it into the decorations array.

I cannot figure this out and I dont understand how I can get at a certain value in a markdown file. For example in the file below, how can I get access to the `/foo/` value in the prose pre component?

<pre>
```js /foo/
function foo() {
  console.log('foo'
}
```
</pre>

and since the prose pre component is generated with three backticks, is there any way I could pass in a custom prop?

**Comment by AbhimanyuAryan** [comment]
interesting I am also looking for an answer to this

**Comment by phat-dang** [comment]
You can use the default ProsePre (https://content.nuxt.com/components/prose#prosepre). 
Source: https://github.com/nuxt-modules/mdc/blob/main/src/runtime/components/prose/ProsePre.vue

Then, custom your template and css
